include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -284,25 +298,25 @@ class cpp_dec_float
    }
 
    // Specific special values.
-   static const cpp_dec_float& nan()
+   static BOOST_CONSTEXPR_OR_CONST cpp_dec_float nan()
    {
-      static const cpp_dec_float val(cpp_dec_float_NaN);
-      init.do_nothing();
-      return val;
+      return cpp_dec_float(cpp_dec_float_NaN);
```

Comment:
I wonder about this - I _think_ what you're really looking for here is `consteval`, but that's a C++20 feature :(  

I'm reasonably sure that the constructor here will be runtime evaluated (ie worst than before!) unless every *use* of the function is marked as `constexpr` which is error prone at best.  I think I'm right in saying that GCC will optimise this to a const-expression when used anyway, but it's not required to, and I'm fairly sure msvc does not do so.

So I think we either make these constants member values:

```
static constexpr cpp_dec_float nan = cpp_dec_float(cpp_dec_float_nan);
```

Which still requires an out of line definition (until C++17 anyway).  Or we do something like:

```
static constexpr const cpp_dec_float& nan()
{
  static constexpr cpp_dec_float val(cpp_dec_float_nan);
   return nan;
}
```